utils/docker_utils.go

```diff
@@ -303,3 +308,59 @@ func getImageConfig(image string) (container.Config, error) {
 	}
 	return config, nil
 }
+
+func getLayersFromManifest(manifestPath string) ([]string, error) {
+	type Manifest struct {
+		Layers []string
+	}
+
+	manifestJSON, err := ioutil.ReadFile(manifestPath)
+	if err != nil {
+		errMsg := fmt.Sprintf("Could not open manifest to get layer order: %s", err)
+		return []string{}, errors.New(errMsg)
+	}
+
+	var imageManifest []Manifest
+	err = json.Unmarshal(manifestJSON, &imageManifest)
+	if err != nil {
+		errMsg := fmt.Sprintf("Could not unmarshal manifest to get layer order: %s", err)
+		return []string{}, errors.New(errMsg)
+	}
+	return imageManifest[0].Layers, nil
+}
+
+func unpackDockerSave(tarPath string, target string) error {
+	if _, ok := os.Stat(target); ok != nil {
+		os.MkdirAll(target, 0777)
+	}
+
+	tempLayerDir := target + "-temp"
+	err := UnTar(tarPath, tempLayerDir)
+	if err != nil {
+		errMsg := fmt.Sprintf("Could not unpack saved Docker image %s: %s", tarPath, err)
+		return errors.New(errMsg)
+	}
+
+	manifest := filepath.Join(tempLayerDir, "manifest.json")
+	layers, err := getLayersFromManifest(manifest)
+	if err != nil {
+		return err
+	}
+
+	for _, layer := range layers {
+		layerTar := filepath.Join(tempLayerDir, layer)
+		if _, err := os.Stat(layerTar); err != nil {
+			glog.Infof("Did not unpack layer %s because no layer.tar found", layer)
```

Comment:
If it's an empty layer it only has a manifest and a metadata json file, but no contents to unpack